pkg/minikube/image/image.go

```diff
@@ -172,19 +173,45 @@ func retrieveImage(ref name.Reference) (v1.Image, error) {
 		return img, nil
 	}
 	// reference does not exist in the local daemon
+	klog.Infof("daemon lookup for %+v: %v", ref, err)
+
+	img, err = retrieveRemote(ref, defaultPlatform)
 	if err != nil {
-		klog.Infof("daemon lookup for %+v: %v", ref, err)
+		return nil, err
 	}
+	return fixPlatform(ref, img, defaultPlatform)
+}
 
-	platform := defaultPlatform
-	img, err = remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithPlatform(platform))
+func retrieveRemote(ref name.Reference, p v1.Platform) (v1.Image, error) {
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain), remote.WithPlatform(p))
 	if err == nil {
 		return img, nil
 	}
 
 	klog.Warningf("authn lookup for %+v (trying anon): %+v", ref, err)
-	img, err = remote.Image(ref)
-	return img, err
+	return remote.Image(ref, remote.WithPlatform(p))
+}
+
+func fixPlatform(ref name.Reference, img v1.Image, p v1.Platform) (v1.Image, error) {
```

Comment:
does this hack still need to be there even with latest https://github.com/google/go-containerregistry ?
if yes I recommend creating an issue with example logs to describe it, so this be fixed in the upstream
and add that issue as a comment in this func so we know the exact reason for this hack and also help other projects to fix the bug for everyone

Reply:
@ilya-zuyev please add the links to the issue for this hack in the comment

Reply:
Done